pkg/bigqueryexporter/bigquery.go

```diff
@@ -0,0 +1,57 @@
+package bigqueryexporter
+
+import (
+	"context"
+	"github.com/openshift/sippy/pkg/db"
+	"github.com/openshift/sippy/pkg/db/models"
+	"gorm.io/gorm/clause"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/api/iterator"
+)
+
+var (
+	projectID = "openshift-ci-data-analysis"
+)
+
+type Client struct {
+	*bigquery.Client
+}
+
+func New(ctx context.Context) (*Client, error) {
+	client := &Client{}
+
+	bqClient, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, err
+	}
+	client.Client = bqClient
+
+	return client, nil
+}
+
+func (client *Client) ExportReleaseTags(ctx context.Context, db *db.DB) error {
+	rows := make([]models.ReleaseTag, 0)
+	query := client.Query(`
+		SELECT
+			ROW_NUMBER() OVER() id, *
+		FROM ` + "`ci_data.ReleaseTags`")
+	it, err := query.Read(ctx)
+	if err != nil {
+		return err
+	}
+
+	for {
+		row := models.ReleaseTag{}
+		err := it.Next(&row)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		rows = append(rows, row)
+	}
+
+	return db.DB.Clauses(clause.OnConflict{UpdateAll: true}).Create(&rows).Error
```

Comment:
This is interesting, on an id conflict it will update the row instead? Could we have an issue with orphaned rows that should be deleted?

Reply:
This lets me fetch all data from bigquery and update the rows if needed. I should probably change this to only fetch data from BQ with an ID > our current local max ID.  That should avoid needing to pull down everything, every refresh.

It's true that if data disappears from bigquery, we will still keep the data in our DB.  Removing data from bigquery seems unlikely in the short term, but I'll see if I can think about a way to fix this.

Reply:
I had it in my head that the db,or most of it, might be transient and could be wiped out and rebuilt. That however may not be a great thing to do on refreshes as it would bring down sippy UI for a bit in the middle. Probably can disregard my concern here.